### DISCUSSION TITLE
Json Formating

### DISCUSSION THREAD

**Discussion by smahad786** [revelc/formatter-maven-plugin#565] [Q&A]
I have a Json Config file that does not have an extension as .json. By default this plugin is not formatting this file. Is there a way to format a json file without json extension?

**Comment by ctubbsii** [comment]
The current version of the plugin is highly dependent on the file extension to detect which formatter it runs for a particular file. This would not be the case if we were to split this plugin into separate mojos, as described in #254, after which you could list the file explicitly for that formatter. However, that work has not been started, so for now, this won't work for your case. To work around this, you could have a task that renames your file after formatting.

**Comment by smahad786** [comment]
Thanks @ctubbsii

**Reply by hazendaz** [reply]
To add, I think we are coded heavily to extension.  If you were to use any number of other plugins that would rename it for you on the fly during build to .json, do the processing then run that again at end that might help.  I think possibly the resource plugin might assist there.  Its a bit hacky approach.  Other thing that is possible is that we could think about an extension file inclusion list.  Not sure what system you work on that requires this but Tandem comes to mind in guardian space usage where extensions are not allowed.  What exactly is the situation that would cause need for no extension to help us understand?

**Reply by ctubbsii** [reply]
We already have an "inclusion" list, but it's not per-formatter. We have file extensions hard-coded per-formatter. Once the formatters are in their own mojos, they can ignore extensions and just use the includes/excludes directly (with a default includes of `**/*.js` or similar, depending on the formatter). The alternative is to create a separate includes/excludes pair of config for each separate formatter, which really sucks.
